### ISSUE TITLE
Bug or unexcepted usage with the thread pool

### ISSUE THREAD

**Issue by homura** [andywer/threads.js#270]
```ts
// main.ts
import { Pool, spawn, Worker } from 'threads';

const pool = Pool(() => spawn(new Worker('./worker')), 1);

const task1 = pool.queue(function(x) {
  return x(1, 1);
});
const task2 = pool.queue(function(x) {
  return x(2, 2);
});
const task3 = pool.queue(function(x) {
  return x(3, 3);
});
const task4 = pool.queue(function(x) {
  return x(4, 5);
});

const wait = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));

async function main() {
  // uncomment follow `await`, and it will never be resolved if the `await` takes longer than tasks
  // await wait(1000);
  await task1.then(console.log);
  await task2.then(console.log);
  await task3.then(console.log);
  await task4.then(console.log);

  await pool.terminate();
}

main();
```

```js
// worker.js
const { expose } = require("threads/worker");

const wait = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

async function add(a, b) {
  // simulate a CPU calculation heavily
  await wait(50);
  return a + b;
}

expose(add);
```

To make it works as expected, I need to keep the `Promise` manually

```ts
// main.ts
import { Pool, spawn, Worker } from 'threads';

const pool = Pool(() => spawn(new Worker('./worker')), 1);

const task1 = pool.queue(function(x) {
  return x(1, 1);
}).then;
const task2 = pool.queue(function(x) {
  return x(2, 2);
}).then;
const task3 = pool.queue(function(x) {
  return x(3, 3);
}).then;
const task4 = pool.queue(function(x) {
  return x(4, 5);
}).then;

const wait = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));

async function main() {
  // now it works, but I can't cancel tasks anymore
  await wait(1000);
  await task1(console.log);
  await task2(console.log);
  await task3(console.log);
  await task4(console.log);

  await pool.terminate();
}

main();
```

**Comment by andywer** [comment]
Hey @homura! Thanks for reporting.

Can you try to monkey-patch the fix I drafted in #271? Haven't had time to try it yet, but that should solve it.

**Comment by andywer** [comment]
(The PR's tests are failing right now, because the error handling needs to be tweaked, but you should be able to check if that solves your particular issue)

**Comment by homura** [comment]
> Hey @homura! Thanks for reporting.
> 
> Can you try to monkey-patch the fix I drafted in #271? Haven't had time to try it yet, but that should solve it.

Yes, it resolved my problem now. Thanks for your quick fix
